R/request_figma.R

```diff
@@ -0,0 +1,84 @@
+#' Create a request for a specified Figma API Endpoint
+#'
+#' @rdname request_figma_endpoint
+#' @inheritParams httr2::request
+#' @inheritParams httr2::req_method
+#' @inheritParams req_figma_template
+#' @inheritParams req_rigma_agent
+#' @inheritDotParams httr2::req_template
+#' @keywords internal
+#' @importFrom httr2 request req_template req_error resp_body_json req_method
+#' @importFrom rlang is_null
+request_figma_endpoint <- function(endpoint = NULL,
+                                   base_url = "https://api.figma.com",
+                                   template = NULL,
+                                   method = NULL,
+                                   token = NULL,
+                                   ...) {
+  req <- httr2::request(base_url) %>%
+    req_figma_template(
+      endpoint = endpoint,
+      template = template,
+      ...
+    ) %>%
+    httr2::req_error(
+      body = function(resp) {
+        body <- httr2::resp_body_json(resp)
+        body[["message"]] %||% body[["err"]]
+      }
+    ) %>%
+    req_rigma_agent(token = token)
+
+  if (rlang::is_null(method)) {
+    return(req)
+  }
+
+  req %>% httr2::req_method(method)
+}
```

Comment:
Great! This is a very nice example of generalization 🚀 
I suggest to keep only this part of the function that is independent of the API at hand. 
We may rename the function to `request_figma_handler()`

```
request_figma_handler <- function(
  req,
  base_url = "https://api.figma.com",
  template = NULL,
  method = NULL,
  token = NULL
) {
  req %>%
    ...
}
```